presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanNodeStats.java

```diff
@@ -153,4 +154,9 @@ public PlanNodeStats mergeWith(PlanNodeStats other)
                 planNodeOutputPositions, planNodeOutputDataSize,
                 operatorInputStats);
     }
+
+    protected void checkMergeable(PlanNodeStats other)
+    {
+        checkArgument(this.getClass() == other.getClass(), "Cannot merge stats %s and %s, make sure all worker nodes have consistent configuration", this, other);
```

Comment:
We don't know `make sure all worker nodes have consistent configuration` is the root cause yet

Reply:
I agree. But we believe it _may_ be the cause, so i think it's a good advice to have